tests/test_freezing/test_freezing_compare.py

```diff
@@ -167,3 +168,80 @@ def test_get_additions(self):
         assert_frame_equal(
             expected_outcome_df, result.reset_index(drop=True)
         ), "get_additions() output is not as expected."
+
+
+class TestCheckForSplitCases:
+    """Tests for check_for_split_cases()."""
+
+    def create_test_additions_df(self) -> pd.DataFrame:
+        """Create a test additions df."""
+        input_cols = ["reference", "period", "instance", "other"]
+        data = [
+            ["A", 202412, 2.0, 1.0],
+            ["B", 202412, None, None],
+            ["C", 202412, 0.0, 1.0],
+            ["D", 202412, 1.0, 2.0],
+            ["E", 202412, None, 4.0],
+        ]
+        input_additions_df = pd.DataFrame(data=data, columns=input_cols)
+        return input_additions_df
+
+    def create_test_amendments_df(self) -> pd.DataFrame:
+        """Create a test amendments df."""
+        input_cols = ["reference", "period", "instance", "other"]
+        data = [
+            ["C", 202412, 0.0, 1.0],
+            ["D", 202412, 1.0, 2.0],
+            ["F", 202412, None, 4.0],
+        ]
+        input_amendments_df = pd.DataFrame(data=data, columns=input_cols)
+        return input_amendments_df
+
+    def create_expected_additions_df(self) -> pd.DataFrame:
+        """Create the expected additions df after split cases are handled."""
+        input_cols = ["reference", "period", "instance", "other"]
+        data = [
+            ["A", 202412, 2.0, 1.0],
+            ["B", 202412, None, None],
+            ["E", 202412, None, 4.0],
+        ]
+        expected_additions_df = pd.DataFrame(data=data, columns=input_cols)
+        return expected_additions_df
+
+    def create_expected_amendments_df(self) -> pd.DataFrame:
+        """Create the expected amendments df after split cases are handled."""
+        input_cols = ["reference", "period", "instance", "other"]
+        data = [
+            ["C", 202412, 0.0, 1.0],
+            ["D", 202412, 1.0, 2.0],
+            ["F", 202412, None, 4.0],
+            ["C", 202412, 0.0, 1.0],
+            ["D", 202412, 1.0, 2.0],
+        ]
+        expected_amendments_df = pd.DataFrame(data=data, columns=input_cols)
+        return expected_amendments_df
+
+    def test_check_for_split_cases(self):
+        """Test for check_for_split_cases()."""
+        # Create test dataframes
+        input_additions_df = self.create_test_additions_df()
+        input_amendments_df = self.create_test_amendments_df()
+        expected_additions_df = self.create_expected_additions_df()
+        expected_amendments_df = self.create_expected_amendments_df()
+
+        # Run the function
+        result_additions_df, result_amendments_df = check_for_split_cases(
+            input_additions_df, input_amendments_df, test_logger
+        )
+
+        # Check the output
+        assert_frame_equal(
+            expected_additions_df, result_additions_df.reset_index(drop=True)
+        ), "check_for_split_cases() additions output is not as expected."
```

Comment:
This may be right, but I don't get it. Why is there a string after the comma?
), "get_additions() output is not as expected."

Reply:
It's a custom error message for if the assert fails :)